buildSrc/src/main/kotlin/test/server/tests/Timeout.kt

```diff
@@ -35,7 +35,7 @@ internal fun Application.timeoutTest() {
                         override val contentType = ContentType.Application.OctetStream
                         override suspend fun writeTo(channel: ByteWriteChannel) {
                             for (offset in response.indices) {
-                                channel.writeFully(response, offset, 1)
+                                channel.writeFully(response, offset, offset + 1)
```

Comment:
Turns out the API changed from this argument being the length to the offset.  Because the test server kinda swallows exceptions, it took me a while to figure out what was going on there 😅

Reply:
It would be nice to implement something to automatically propagate exceptions thrown on the test server. What do you think?

Reply:
Adding exception handling has uncovered a test that's been silently failing for a while. 😅

Reply:
Oh, there is an ancient bug that prevents me from fixing this. https://youtrack.jetbrains.com/issue/KTOR-633/How-to-change-io.ktor.http.cio.CIOMultipartDataBase.formFieldLimit

![](https://i.kym-cdn.com/photos/images/newsfeed/001/515/121/dfd.jpg)

Reply:
Oh nm, I got burned by the already-running test server.

Reply:
"20 minutes adventure" :D

Reply:
I've just discovered `ErrorCollector` + `EngineTestBase.createServer`, which is pretty close to what I meant under "automatically propagate exceptions thrown on the test server". But I thought about a standard mechanism for all users of `testApplication` DSL.

In other words, make this test fail by default:
```kotlin
@Test
fun `failure on a server`() = testApplication {
    routing {
        get("/") { fail() }
    }

    client.get("/")
}
```

Reply:
It is not related to the test server, so I will move this conversation into Slack